testmodule/fileparser.py

```diff
@@ -248,6 +273,53 @@ def check_F(job_file, group, number):
             pass
 
 
+def check_G(job_file, group, number):
+    """Check (JBI-W7).
+
+    If foldername is MAIN, the command CALL JOB:SET_IDS_FULL (with arguments) must be
+    present and called before CALL JOB:TRIGGER ARGF"PROGRAMM_EIN"
+
+    Parameters
+    ----------
+    job_file : ojb:`jobFile`
+        Object of a jobFile class.
+    group : str
+        Group of the warning.
+    number : int
+        Number of the warning.
+
+    Returns
+    -------
+    warnings : array
+        Error messages.
+    """
+    index_set = []
+    index_trigger = []
+
+    if job_file.foldername == "MAIN":
+        index_set = [
+            (i, line.strip())
+            for i, line in enumerate(job_file.lines)
+            if line.startswith("CALL JOB:SET_IDS_FULL")
+        ]
+        index_trigger = [
+            (i, line.strip())
+            for i, line in enumerate(job_file.lines)
+            if line.startswith('CALL JOB:TRIGGER ARGF "PROGRAMM_EIN"')
```

Comment:
the strings of the relevant lines `'CALL JOB:TRIGGER ARGF "PROGRAMM_EIN"'` and `CALL JOB:SET_IDS_FULL` should be implemented as variables in the function to make it easier to adjust them in all places at once